modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -0,0 +1,256 @@
+                                        "The number of state variables this UMAT is going to use");
+  return params;
+}
+
+AbaqusUMATStress::AbaqusUMATStress(const InputParameters & parameters)
+  : ComputeStressBase(parameters),
+    _plugin(getParam<FileName>("plugin")),
+    _aqNSTATV(getParam<unsigned int>("num_state_vars")),
+    _aqSTATEV(_aqNSTATV),
+    _aqPROPS(getParam<std::vector<Real>>("constant_properties")),
+    _aqNPROPS(_aqPROPS.size()),
+    _stress_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "stress")),
+    _total_strain(getMaterialProperty<RankTwoTensor>(_base_name + "total_strain")),
+    _strain_increment(getMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
+    _jacobian_mult(declareProperty<RankFourTensor>(_base_name + "Jacobian_mult")),
+    _Fbar(getMaterialPropertyOld<RankTwoTensor>(_base_name + "deformation_gradient")),
+    _Fbar_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "deformation_gradient")),
+    _state_var(declareProperty<std::vector<Real>>(_base_name + "state_var")),
+    _state_var_old(getMaterialPropertyOld<std::vector<Real>>(_base_name + "state_var")),
+    _elastic_strain_energy(declareProperty<Real>(_base_name + "elastic_strain_energy")),
+    _plastic_dissipation(declareProperty<Real>(_base_name + "plastic_dissipation")),
+    _creep_dissipation(declareProperty<Real>(_base_name + "creep_dissipation")),
+    _material_timestep(declareProperty<Real>(_base_name + "material_timestep_limit"))
+{
+#ifndef METHOD
+#error "The METHOD preprocessor symbol must be supplied by the build system."
+#endif
+  _plugin += std::string("-") + QUOTE(METHOD) + ".plugin";
+
+  // Read mesh dimension and size UMAT arrays (we always size for full 3D)
+  _aqNTENS = 6; // Size of the stress or strain component array (NDI+NSHR)
+  _aqNSHR = 3;  // Number of engineering shear stress components
+  _aqNDI = 3;   // Number of direct stress components (always 3)
+
+  _aqDDSDDT.resize(_aqNTENS);
+  _aqDRPLDE.resize(_aqNTENS);
+  _aqSTRAN.resize(_aqNTENS);
+  _aqDFGRD0.resize(9);
+  _aqDFGRD1.resize(9);
+  _aqSTRESS.resize(_aqNTENS);
+  _aqDDSDDE.resize(_aqNTENS * _aqNTENS);
+  _aqDSTRAN.resize(_aqNTENS);
+
+  // Open the library
+#ifdef LIBMESH_HAVE_DLOPEN
+  _handle = dlopen(_plugin.c_str(), RTLD_LAZY);
+
+  if (!_handle)
+    paramError("plugin", "Cannot open library: ", dlerror());
+
+  // Reset errors
+  dlerror();
+
+  // Snag the function pointer from the library
+  {
+    void * pointer = dlsym(_handle, "umat_");
+    _umat = *reinterpret_cast<umat_t *>(&pointer);
+  }
+
+  // Catch errors
+  const char * dlsym_error = dlerror();
+  if (dlsym_error)
+  {
+    dlclose(_handle);
+    paramError("plugin", "Cannot load symbol 'umat_': ", dlsym_error);
+  }
+#else
+  paramError("plugin",
+             "AbaqusUMATStress requires an operating system with support for the POSIX function "
+             "'dlopen' to dynamically load UMAT plugins.");
```

Comment:
I wouldn't think this is a `paramError`

Reply:
The reason I'm using param error is to get a more user friendly output with a position in the input file. Also, the plugin parameter specifies the file that should be dynamically loaded but cannot.